evaluator.fhir/src/test/java/org/opencds/cqf/cql/evaluator/engine/retrieve/BundleRetrieveProviderTests.java

```diff
@@ -183,6 +203,23 @@ public void test_filterToValueSet() {
         assertThat(results, is(iterableWithSize(1)));
     }
 
+    @Test
+    public void test_filterToValueSet_codePathPointsToChoiceTypeWithReference() {
+        IBaseBundle bundle = this.loadBundle("TestBundleMedicationAdministrationWithReference.json");
+        BundleRetrieveProvider brp = new BundleRetrieveProvider(fhirContext, bundle);
+
+        // Despite the Medication referenced by MedicationAdministration matching the code, the
+        // RetrieveProvider (as it exists today) does not support resolving the reference and then
```

Comment:
Hi @aaron-nash! Thanks for the contribution here. I appreciate the comment explaining the code and test. Would you mind updating/removing this comment? Once this code is merged the comment will be out of date since the BundleRetrieveProvider will have support added.

Reply:
Hi @JPercival. Thanks for the comment. I don't fully follow what you're asking. If this PR is merged, it will still be the case that `BundleRetrieveProvider` does not support resolving a reference in the `codePath` and matching the resolved reference against the code. In this test case `results` would not be empty if the retrieve provider supported that. What am I missing?

Reply:
Please let me know if any changes are still required.